packages/core/src/agent/openai/worker.ts

```diff
@@ -0,0 +1,403 @@
+// Assuming that the necessary interfaces and classes (like BaseTool, OpenAI, ChatMessage, CallbackManager, etc.) are defined elsewhere
+
+import { randomUUID } from "crypto";
```

Comment:
Blocker: This is a NodeJS specific import correct? If so we should import from `"node:crypto"` to not break the other bundles.